docs/proposals/11-scheduling.md

```diff
@@ -0,0 +1,133 @@
+
+### Non-Goals
+- The scheduler should not act on `Machine` updates like `NetworkInterface` attachments
+
+## Proposal
+
+For every created `Machine` a resource `Reservation` object will be created. The `poollet` will continue to broker only `Machine`s with a `.spec.machinePoolRef` set. The `Reservation` is a condensed derivation of a `Machine` containing the requested resources like `NetworkInterfaces` and the attached `MachineClass` resource list. The `IRI` will be extended to be able to broker the `Reservation`s. Once the `Reservation` hits a pool provider, the decision can be made if the `Reservation` can be fulfilled or not. In case of accepting the `Reservation`, resources needs to be blocked until a) the reservation is being deleted b) the corresponding `Machine` is being placed on the pool provider. The status of the `Reservation` is being propagated up to the layer where it was created. After some time it contains a list of possible pools where the scheduler can pick one, set the `Machine.spec.machinePoolRef` and the `poollet` will broker the `Machine` to the next level.
+
+`Reservation` resource: 
+```
+apiVersion: core.ironcore.dev/v1alpha1
+kind: Reservation
+metadata:
+  namespace: default
+  name: reservation
+spec:
+  # Pools define each pool where resources should be reserved
+  pools: 
+    - poolA
+    - poolB
+    - poolC
+  # Resources defines the requested resources in a ResourceList
+  resources:
+    localStorage: 10G
+    nics: 2
+    cpu: 2
+status:
+  # Pools describe where the resource reservation was possible
+  pools: 
+    - name: poolA
+      rating: 2
+      state: accepting
+    - name: poolB
+      ratring: 1
+      state: rejecting
+```
+
+Added `IRI` methods:
+
+```protobuf
+rpc ListReservations(ListReservationsRequest) returns (ListReservationsResponse) {};
+rpc CreateReservation(CreateReservationRequest) returns (CreateReservationResponse) {};
+rpc DeleteReservation(DeleteReservationRequest) returns (DeleteReservationResponse) {};
+```
+
+
+### Detailed flow
+1. If `.spec.machinePoolRef` is not set, the scheduler creates a `Reservation` that includes the required resources.
+2. `poollet`s which match `.spec.pools` of `Reservation` pick it up and broker it one layer down
+3. `poriver` evaluates the `Reservation` and sets its state to `accepting` or `rejecting`, which is then propagated up the hierarchy.
+4. The scheduler will pick a `pool` on every layer and update the `.spec.machinePoolRef` of the related `Machine`
+5. Once a `Machine` reaches a `provider` with a relating `Reservation`, the `Reservation` will be replaced through 
+   the `Machine`. Deletion process of `Reservation` is initiated through setting state to `releasing`, which is 
+   propagated up.
+6. Scheduler will delete top `Reservation` which triggers deletion of all brokered `Reservation`s 
+
+
+
+### Advantages
+- `Reservation`s can also be used to block resources for a specific use-case
```

Comment:
Without creating a `Machine` object?

Reply:
i would imagine this can be used for system components other than `Machine`s. Though I am not sure if other sub-systems would be a better fit to leverage this.